addons/pos_restaurant/static/src/app/control_buttons/control_buttons.js

```diff
@@ -27,6 +27,13 @@ patch(ControlButtons.prototype, {
         this.dialog.add(NumberPopup, {
             startingValue: this.currentOrder?.getCustomerCount() || 0,
             title: _t("Guests?"),
+            feedback: (buffer) => {
+                if (buffer && buffer.length > 0) {
+                    const guestCount = parseInt(buffer, 10) || 0;
+                    this.currentOrder.setCustomerCount(guestCount);
```

Comment:
you should never mutate in the feedback.
Also you probably want the return to be something like ``value ?  `Amount per customer: ${value}` : '' ``

Reply:
Sure, my bad for the earlier oversight on mutating data in the feedback function, which has now been corrected.

Regarding the return value, the `this.env.utils.formatCurrency` function manages the currency symbol. Changes in XML were made to align with the specified positioning and supporting text requirements.